RealEstateAgencyApp.Infrastructure/Persistence/DBSeeeder.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RealEstateAgencyApp.Domain.DataSeeders;
+
+namespace RealEstateAgencyApp.Infrastructure.Persistence;
+
+/// <summary>
+/// Provides methods for seeding the database with initial test data.
+/// Contains separate methods for seeding each entity type and a combined method for all data.
+/// </summary>
+public static class DbSeeder
+{
+    /// <summary>
+    /// Seeds the database with real estate objects if the table is empty.
+    /// Also resets the auto-increment counter to continue from the next available ID.
+    /// </summary>
+    /// <param name="context">The database context to seed data into.</param>
+    /// <returns>A task that represents the asynchronous operation.</returns>
+    public static async Task SeedEstatesAsync(DBContext context)
+    {
+        if (!context.RealEstateObjects.Any())
```

Comment:
Тут и дальше по коду:
```suggestion
        if (!await context.RealEstateObjects.AnyAsync())
```